File: Java/SETPractice/src/main/java/SOLIDPrinciple/DependencyInversion/EmployeeServiceLayer.java
package SOLIDPrinciple.DependencyInversion;

public class EmployeeServiceLayer {
    IEmployeeDataAccessLayer DataAccessLayer;

    public EmployeeServiceLayer() {
        DataAccessLayer = DataAccessFactory.GetEmployeeDataAccessLayerObj();
    }

    public Employee GetEmployeeDetails(int id) {
        return DataAccessLayer.GetEmployee(id);
    }
}

File: DotNet/SETPractice/SOLIDPrinciples/DependancyInversion/EmployeeServiceLayer.cs
namespace SOLIDPrinciples.DependancyInversion
{
    public class EmployeeServiceLayer
    {
        public IEmployeeDataAccessLayer DataAccessLayer { get; set; }

        public EmployeeServiceLayer() 
        {
            DataAccessLayer = DataAccessFactory.GetEmployeeDataAccessLayerObj();
        }

        public Employee GetEmployeeDetails(int id)
        {
            return DataAccessLayer.GetEmployee(id);
        }
    }
}
